trajopt/include/trajopt/problem_description.hpp

```diff
@@ -249,6 +251,15 @@ struct UserDefinedTermInfo : public TermInfo
   /** @brief The user defined jocobian function */
   sco::MatrixOfVector::func jacobian_function;
 
+  /** @brief If added as a cost it will use this penalty type */
+  sco::PenaltyType cost_penalty_type{sco::PenaltyType::SQUARED};
+
+  /** @brief If added as a constraint it will use this constraint type */
+  sco::ConstraintType constraint_type{sco::ConstraintType::EQ};
+
+  /** @brief Coefficients for user defined error function */
+  Eigen::VectorXd coeff;
```

Comment:
Not sure, there is no way to know the size of the error function ahead of time. Could send random joint value to error function to get the number of errors. Or could modify CostFromErrFunc::convex to check if size of coeff vector is one it is evaluated as the coeff for each error. Any thoughts?

Reply:
I think your last suggestion is what we are (or at least used to be) doing for joint velocity costs. If size 1 it uses that for all, size dof and it applies it to each joint. Otherwise it errors.

Reply:
Per our discussion I will leave it as is.